### PR TITLE
vdk-jupyter: add UI element indicating a VDK operation is running

### PR DESCRIPTION
**Pull request by yonitoo**
# Why
At the moment there is no way to know if a VDK operation is running until it finishes and shows a success message or error.

# What
Add a status button that pops up when a VDK operation is started and is removed when the operation finishes. When the button is clicked, a dialog shows up with more information about the current running operation.

Status button
Hover
Pop-up dialog

Signed-off-by: Yoan Salambashev <ysalambashev@vmware.com>

### CODE REVIEW

**Comment by duyguHsnHsn** [comment]
Can you add a picture of the button and how it looks in the toolbar

**Comment by duyguHsnHsn** [comment]
The button looks a little bit old styled can you change it a little ?

**Comment by DeltaMichael** [comment]
This seems like it should be a loading wheel/status bar + text info + timer, or something similar. 

E.g.

Screenshot 2023-08-03 at 14 33 36

**Comment by sabadzhiev** [comment]
Visually it looks good (exactly what we discussed on the daily). 

However will leave the other team members to review the code and approve.

**Comment by duyguHsnHsn** [comment]
The only thing i do not like about the button is the borderline, it still seems old fashioned. Everything else looks fine

File: projects/vdk-plugins/vdk-jupyter/vdk-jupyterlab-extension/package-lock.json
Comment (duyguHsnHsn) [inline projects/vdk-plugins/vdk-jupyter/vdk-jupyterlab-extension/package-lock.json:1]: be sure that the lock file is compatible with other late changes

File: projects/vdk-plugins/vdk-jupyter/vdk-jupyterlab-extension/src/components/StatusButton.tsx [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+      clearInterval(this.timerId);
+      this.timerId = undefined;
+    }
+    const timeElement = this.buttonElement.querySelector('span');
+    if (timeElement) {
+      timeElement.innerHTML = '00:00:00';
+    }
+  }
+
+  private formatTime(seconds: number): string {
```
Comment (duyguHsnHsn) [inline projects/vdk-plugins/vdk-jupyter/vdk-jupyterlab-extension/src/components/StatusButton.tsx:86]: Why are you making your own functions for this? Aren't there existing tools or libraries that already do that for you?

**Review by duyguHsnHsn — APPROVED** [review]
Looks good please check the comments I left!

File: projects/vdk-plugins/vdk-jupyter/vdk-jupyterlab-extension/src/components/StatusButton.tsx (hunk 1 above)
Comment (yonitoo) [reply projects/vdk-plugins/vdk-jupyter/vdk-jupyterlab-extension/src/components/StatusButton.tsx:86]: Not sure, I will check for such.

Comment (yonitoo) [reply projects/vdk-plugins/vdk-jupyter/vdk-jupyterlab-extension/src/components/StatusButton.tsx]: Changed the manual approach to a built-in one.

File: projects/vdk-plugins/vdk-jupyter/vdk-jupyterlab-extension/package-lock.json
Comment (yonitoo) [reply projects/vdk-plugins/vdk-jupyter/vdk-jupyterlab-extension/package-lock.json:1]: Done.

**Comment by murphp15** [comment]
looks good

**Review by murphp15 — APPROVED** [review]

**Pull request merged**